### DISCUSSION TITLE
Console.Log working, but req.json() only responding some times in custom endpoint.

### DISCUSSION THREAD

**Discussion by TomS-** [directus/directus#9875] [Archive]
```js
    router.post('/file', async (req, res, next) => {
        const busboy = new Busboy({ headers: req.headers });

        try {
            busboy.on('file', (fieldname, file, filename, encoding, mimetype) => {
                file.on('data', async data => {
                    const options = {
                        'method': 'POST',
                        'url': 'https://api.fusioo.com/v3/files',
                        'headers': {
                            'Authorization': `Bearer ${token}`
                        },
                        formData: {
                            'file': {
                                'value': data,
                                'options': { 'filename': filename, 'contentType': mimetype }
                            }
                        }
                    };
                    
                    request(options, (error, response) => {
                        if(error) console.log(error);
                        console.log(response.body);
                        req.json(response.body);
                        next();
                    });
                });
            });

            busboy.on('finish', () => next());
            req.pipe(busboy);

        } catch (error) {
            next(new InvalidQueryException(error.message));
        };
    });
```

Hey guys,
I have a file upload that I sent to Directus to upload to our CRM, I do it to Directus first because the CRM only has one type of access and that's full access so I want to hide the auth token. I'm expecting the file ID back which appears in the console.log and is sometimes returned back to the browser. But other times I get a 400 error, while in the console I get the correct response `{"code":10000,"message":"Success","data":{"id":"ib83981d6c55f4de38b2a8c697055ca54","title":"Our_Approach.jpg"}`.

Does anyone know how to correct this?

**Comment by rijkvanzanten** [comment]
> Does anyone know how to correct this?

I think it's because you're calling `next()` _before_ your `request` is done. Note how you're calling the express callback telling to finish the response both in the callback of `request` (which is correct), but also in when busboy finishes.

It looks to me like busboy finishes reading the file, and calls express' `next()`, _while_ your request is still going out to the internet. Once your request finishes, express already responded, as `next()` was called in busboy's `finish`
